std/json.d

```diff
@@ -1110,6 +1165,91 @@ if (isInputRange!T && !isInfinite!T && isSomeChar!(ElementEncodingType!T))
     assert(json["key1"]["key2"].integer == 1);
 }
 
+version(unittest)
+{
+    struct OrderedAA(Key, Value)
```

Comment:
I think others (not me...) were concerned about exposing json with ordered_fields, but I didn't see objections to exposing OrderedAA in std.array; otherwise this'll require someone who wants to use json with ordered_fields to reimplement `OrderedAA` below, which is not DRY and much worse than exposing json with ordered_fields. In short: using json with ordered_fields should be as simple as:
```
import std.array:OrderedAA;
static struct OrderedAAPolicy
{
alias ObjectType = OrderedAA;
enum sortObjectMembers = false;
}

auto json1 = parseJSONTemplate!OrderedAAPolicy(...);

```

Reply:
Agreed.  However, now that `std.json` does not depend on having `OrderedAA` defined, these can be 2 separate changes.  When I make a separate PR to add `OrderedAA` to `std.array`, we can remove this instance I've included in the unittest and swap it out for that.

Note that by separating these changes into distinct PRs, it makes this one easier to accept on it's own.  Once this one gets accepted, we can start using this feature in `dmd` by simply including our own definition of `OrderedAA` until we can add it to `std.array`.  I'm guessing that adding `OrderedAA` is going to be a fair amount of work with lots of revision and documentation so I would rather not have this change depend on that one.